eth/state_accessor.go

```diff
@@ -93,7 +103,7 @@ func (eth *Ethereum) hashState(ctx context.Context, block *types.Block, reexec u
 		if !readOnly {
 			statedb, err = state.New(current.Root(), database, nil)
 			if err == nil {
-				return statedb, func() { database.TrieDB().Close() }, nil
+				return statedb, noopReleaser, nil
```

Comment:
same question

Reply:
here also if state is available then it was read from disk + same as above we don't need to close the triedb so we can remove the diff.